Extensions/ArtifactEngine/Engine/cilogger.ts

```diff
@@ -0,0 +1,21 @@
+var tl = require('vsts-task-lib');
+
+const area: string = 'artifact-engine';
+
+function getDefaultProps() {
+    return {
+        releaseuri: tl.getVariable('Release.ReleaseUri'),
+        releaseid: tl.getVariable('Release.ReleaseId'),
+        builduri: tl.getVariable('Build.BuildUri'),
+        buildid: tl.getVariable('Build.Buildid')
+    };
+}
+
+export function publishEvent(feature, properties: { [key: string]: any }): void {
+    try {
+        tl.assertAgent('2.125.0');
+        tl.publishTelemetry(area, feature, Object.assign(getDefaultProps(), properties));
+    } catch (err) {
+        tl.debug('Unable to publish telemetry due to lower agent version.');
```

Comment:
since publishTelemetry is also within try block, suggest keeping the log message generic:

Unable to publish telemetry, error: {err}. I assume the error message will include detail of agent version if tl.assertAgent() throws error. if not, let's log agent version also along with the message, like:

Unable to publish telemetry, error: {err}, agentVersion: {agentVersion}